drivers/gpio/sunxi-gpio.c

```diff
@@ -31,10 +31,12 @@
 #define INT_CONTROL_REG(port) (0x0210 + (port) * 0x20)
 #define INT_STATUS_REG(port)  (0x0214 + (port) * 0x20)
 
-static bool
-sunxi_gpio_read_pin(struct device *dev, uint8_t pin)
+static int
+sunxi_gpio_read_pin(struct device *dev, uint8_t pin, bool *value)
 {
-	return mmio_read32(dev->regs + DATA_REG(pin)) & BIT(PIN_INDEX(pin));
+	*value = mmio_read32(dev->regs + DATA_REG(pin)) & BIT(PIN_INDEX(pin));
+
+	return SUCCESS;
```

Comment:
shouldn't this return value instead of success?

Reply:
The value is stored in the value parameter, which is a pointer.

Reply:
It used to, but the API wasn't flexible enough. Some other GPIO controllers may fail to read, and all GPIO controllers have to match the same API. Since there's no failure cases for MMIO, this one just returns the `SUCCESS` error code always. The pointer is used to pass back the value.